plugins/modules/foreman_templates_import.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2020 Anton Nesterov (@nesanton)
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: foreman_templates_import
+short_description: Sync templates from a repository
+description:
+  - Sync provisioning templates, report_templates, partition tables and job templates from external git repository and/or file system.
+  - Based on foreman_templates plugin U(https://github.com/theforeman/foreman_templates).
+  - Some defaults can be set in TemplateSync settings using M(foreman_setting) or GUI.
+  - Module attempts to be idempotent as much as the plugin allows.
+author:
+  - "Anton Nesterov (@nesanton)"
+options:
+  prefix:
+    description: Adds specified string to beginning of the template, but only if the template name does not start with the prefix already.
+    required: false
+    type: str
+  associate:
+    description: Associate to OSes, Locations and Organizations based on metadata.
+    required: false
+    type: str
+    choices:
+     - always
+     - new
+     - never
+  verbose:
+    description: Add template reports to the output.
+    required: false
+    type: bool
+  force:
+    description: Update templates that are locked.
+    required: false
+    type: bool
+  lock:
+    description: Lock imported templates.
+    required: false
+    type: bool
+  branch:
+    description: Branch in Git repo.
+    required: false
+    type: str
+  repo:
+    description: Filesystem path or repo (with protocol), for example /tmp/dir or git://example.com/repo.git or https://example.com/repo.git.
+    required: false
+    type: str
+  filter:
+    description:
+      - Sync only templates with name matching this regular expression, after I(prefix) was applied.
+      - Case-insensitive, snippets are not filtered.
+    required: false
+    type: str
+  negate:
+    description: Negate the prefix.
```

Comment:
What does that mean? Will it remove the `prefix` param from the names?

Reply:
Omg, it negates the filter not the prefix. Wich starts to make sense ;) I'll fix it